pyvista/core/utilities/features.py

```diff
@@ -91,14 +96,167 @@ def voxelize(mesh, density=None, check_surface=True):
     ugrid = pyvista.UnstructuredGrid(grid)
```

Comment:
@bsburnham `flake8` says that `ugrid` is not used anymore. Are you sure to remove this?

Reply:
@tkoyama010 the `extract_points` method returns an `UnstructuredGrid`, so I'm not sure we need to create an empty `ugrid` variable. Unless I've missed something in the documentation?

Reply:
@bsburnham As far as I can see, the changes regarding the voxelize function would be easier to understand if they were reverted in this PR. Note that the function we are trying to add is isolated to voxelize_volume.
By the way, have you checked the file changes at the far right of the tab? It is very useful to know what you are trying to change in this PR.
![image](https://github.com/pyvista/pyvista/assets/7513610/a61680fb-8adc-411f-8d41-1513cbc9c2e2)